src/server/serve.rs

```diff
@@ -435,6 +437,50 @@ fn parse_resolv_conf(content: &str) -> AardvarkResult<Vec<SocketAddr>> {
     Ok(nameservers)
 }
 
+async fn monitor_upstream_resolvers(nameservers: Arc<Mutex<Vec<SocketAddr>>>) {
+    let mut inotify = match Inotify::init() {
+        Ok(inotify) => inotify,
+        Err(err) => {
+            error!("Failed to initialize inotify: {err}");
+            warn!("Nameservers will not be updated on resolv.conf change");
+            return;
+        }
+    };
+    if let Err(err) = inotify
+        .watches()
+        .add(
+            "/etc/resolv.conf",
+            WatchMask::MODIFY,
```

Comment:
it makes sense to put `/etc/resolv.conf` into a const and then use it here and in get_upstream_resolvers()

And I don't think we want to listen on modify, if I look into dnsmasq it calls
`inotify_add_watch(daemon->inotifyfd, path, IN_CLOSE_WRITE | IN_MOVED_TO);` which make sense. We would not want to try to read partial changes if we can avoid it only when the file is closed. And I assume we also must handle th emove case because in most cases to be "atomic" most implementations that chnage the file with actually create a tmpfile and then rename it instead.